File: gmall-realtime/src/test/java/test2.java

/**
 * @ClassName gmall-parent-test2
 * @Author Holden_—__——___———____————_____Xiao
 * @Create 2021年11月30日10:20 - 周二
 * @Describe
 */
public class test2 {
    public static void main(String[] args) throws InterruptedException {
        long num = 1;
        while (true) {
            System.out.println("数据来了" + num++);
            Thread.sleep((int) (Math.random() * (2000 - 100 + 1) + 100));
        }
    }
}

File: gmall-realtime/src/main/scala/com/atguigu/test/Test2.scala
package com.atguigu.test

import org.apache.spark.SparkConf
import org.apache.spark.streaming.dstream.ReceiverInputDStream
import org.apache.spark.streaming.{Seconds, StreamingContext}

/**
 * @ClassName gmall-parent-Test2
 * @Author Holden_—__——___———____————_____Xiao
 * @Create 2021年11月30日13:50 - 周二
 * @Describe
 */
object Test2 {
  def main(args: Array[String]): Unit = {
    val s1: Option[Int] = Some(5)
    val s2: Option[Int] = None

    println(s1.isDefined)
    println(s2.isDefined)
  }
}
